jobs/pacman-awsrules/src/main/java/com/tmobile/cloud/awsrules/cloudtrail/DenyPublicAccessS3BucketStoresCloudTrailLogs.java

```diff
@@ -0,0 +1,150 @@
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, express or
+ * implied. See the License for the specific language governing permissions and
+ * limitations under the License.
+ ******************************************************************************/
+package com.tmobile.cloud.awsrules.cloudtrail;
+import com.amazonaws.util.StringUtils;
+import com.google.common.collect.HashMultimap;
+import com.google.gson.*;
+import com.nimbusds.oauth2.sdk.util.MapUtils;
+import com.tmobile.cloud.awsrules.utils.PacmanUtils;
+import com.tmobile.cloud.awsrules.utils.RulesElasticSearchRepositoryUtil;
+import com.tmobile.cloud.constants.PacmanRuleConstants;
+import com.tmobile.pacman.commons.PacmanSdkConstants;
+import com.tmobile.pacman.commons.exception.InvalidInputException;
+import com.tmobile.pacman.commons.exception.RuleExecutionFailedExeption;
+import com.tmobile.pacman.commons.policy.Annotation;
+import com.tmobile.pacman.commons.policy.BasePolicy;
+import com.tmobile.pacman.commons.policy.PacmanPolicy;
+import com.tmobile.pacman.commons.policy.PolicyResult;
+import com.tmobile.pacman.commons.utils.CommonUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.slf4j.MDC;
+
+import java.util.*;
+
+@PacmanPolicy(key = "deny-public-access-for-s3-bucket-that-stores-cloudtrail-logs", desc = "This rule checks S3 bucket that stores cloudtrail Logs should not be publicly accessible", severity = "critical", category = PacmanSdkConstants.SECURITY)
+public class DenyPublicAccessS3BucketStoresCloudTrailLogs extends BasePolicy {
+    private static final Logger logger = LoggerFactory.getLogger(DenyPublicAccessS3BucketStoresCloudTrailLogs.class);
+
+    @Override
+    public PolicyResult execute(final Map<String, String> ruleParam, Map<String, String> resourceAttributes) {
+
+        MDC.put(PacmanRuleConstants.EXECUTION_ID, ruleParam.get(PacmanRuleConstants.EXECUTION_ID));
+        MDC.put(PacmanRuleConstants.RULE_ID, ruleParam.get(PacmanSdkConstants.POLICY_ID));
+
+        String severity = ruleParam.get(PacmanRuleConstants.SEVERITY);
+        String category = ruleParam.get(PacmanRuleConstants.CATEGORY);
+
+        if (MapUtils.isNotEmpty(ruleParam) && !PacmanUtils.doesAllHaveValue(severity, category)) {
+            logger.info(PacmanRuleConstants.MISSING_CONFIGURATION);
+            throw new InvalidInputException(PacmanRuleConstants.MISSING_CONFIGURATION);
+        }
+
+        String esUrl = CommonUtils.getEnvVariableValue(PacmanSdkConstants.ES_URI_ENV_VAR_NAME);
+        String url = CommonUtils.getEnvVariableValue(PacmanSdkConstants.ES_URI_ENV_VAR_NAME);
+        if (!StringUtils.isNullOrEmpty(url)) {
+            esUrl = url + "/aws/s3/_search";
+        }
+
+        String s3BucketName = resourceAttributes.get("s3bucketname");
+        String accountId = resourceAttributes.get(PacmanRuleConstants.ACCOUNTID);
+        boolean isPublicAccessEnabled = false;
```

Comment:
it's used only inside next`if`, correct?

Reply:
and two vars above as well

Reply:
Addressed in commit